### PR TITLE
Clarify that assertString/Number allow empty string and zero

### PR DESCRIPTION
**Pull request by dvoytenko**
No logic changes. Only documentation and tests updated for edge cases.

### CODE REVIEW

File: src/log.js [hunk 1]
```diff
@@ -328,13 +328,14 @@ export class Log {
...
+   * Throws an error if the first argument isn't a number. The number can
+   * be `0`.
    *
-   * Otherwise see `assert` for usage
+   * For more details see `assert`.
    *
    * @param {*} shouldBeNumber
    * @param {string=} opt_message The assertion message
-   * @return {number} The value of shouldBeTrueish.
+   * @return {number} The number value. Can be `0`.
```
Comment (jridgewell) [inline src/log.js]: I think calling out `NaN` here would be better than `0`.

Comment (dvoytenko) [inline src/log.js]: Good point. Add NaN to the docs, so both are called out now.
